### PR TITLE
List published files for a branch

### PR DESCRIPTION
**Pull request by jmhooper**
This PR adds a new feature that allows users to see what files have been published for a given branch. Work to be done:

- [x] Augment S3PublishedFilesLister to list published files for a specific branch
- [x] Add an endpoint to the published file API to have a `findOne` action that displays files for the branch
- [x] Add a React component that leverages the new API action and displays the list of files

### CODE REVIEW

**Comment by jmhooper** [comment]
Checked all the boxes. This PR is not longer WIP and is review ready.

File: public/swagger/index.yml [hunk 1]
```diff
@@ -318,6 +318,12 @@ paths:
           schema:
             $ref: "Error.json"
   /site/{site_id}/published-branch:
+    parameters:
+    - name: id
```
Comment (brandocalrissian) [inline public/swagger/index.yml]: Shouldn't this be named site_id to line everything up?

File: public/swagger/index.yml [hunk 2]
```diff
@@ -331,6 +337,29 @@ paths:
           description: Not authorized
           schema:
             $ref: "Error.json"
+  /site/{site_id}/published-branch/{branch}:
+    parameters:
+    - name: id
```
Comment (brandocalrissian) [inline public/swagger/index.yml]: id -> site_id ?

File: test/api/unit/services/S3PublishedFileLister.test.js [hunk 3]
```diff
@@ -37,4 +37,58 @@ describe("S3PublishedFileLister", () => {
...
+
+      factory.site({ defaultBranch: "master" }).then(model => {
+        site = model
+        return S3PublishedFileLister.listPublishedFilesForBranch(site, "preview")
+      }).then(publishedFiles => {
+        expect(publishedFiles).to.deep.equal(["abc", "abc/def", "ghi"])
+        done()
+      }).catch(done)
+    })
+  })
```
Comment (brandocalrissian) [inline test/api/unit/services/S3PublishedFileLister.test.js]: Can we add a test where S3.listObjects fails for completeness? We don't cover the error case.

File: test/frontend/components/site/sitePublishedBranch.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        { name: "preview", viewLink: "www.example.gov/preview", site: { id: 1 }, files: incorrectFiles },
+        { name: "master", viewLink: "www.example.gov/wrong", site: { id: 2 }, files: incorrectFiles },
+      ]
+    }
+
+    const wrapper = shallow(<SitePublishedBranch {...props} />)
+    expect(wrapper.find("table")).to.have.length(1)
+    expect(wrapper.find("table").contains("abc")).to.be.true
+    expect(wrapper.find("table").contains("abc/def")).to.be.true
+    expect(wrapper.find("table").contains("ghi")).to.be.true
```
Comment (brandocalrissian) [inline test/frontend/components/site/sitePublishedBranch.test.js]: Note my inconsistency-- I'm not griping about these string literals.

:shrug:

**Pull request merged**